src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -64,17 +63,15 @@ public class JsonRpc : IDisposableObservable
         /// </summary>
         private readonly Func<Task, object, JsonRpcMessage> handleInvocationTaskResultDelegate;
 
-        /// <summary>
-        /// A dictionary to map clr method names to <see cref="JsonRpcMethodAttribute" /> values.
-        /// </summary>
-        private readonly ReadOnlyDictionary<string, string> requestMethodToClrMethodMap;
+        private readonly List<Tuple<object, ReadOnlyDictionary<string, string>>> targetRequestMethodToClrMethodMap;
 
         private readonly CancellationTokenSource disposeCts = new CancellationTokenSource();
 
         private Task readLinesTask;
         private int nextId = 1;
         private bool disposed;
         private bool hasDisconnectedEventBeenRaised;
+        private bool startedListening = false;
```

Comment:
FxCop whines when you initialize fields to their default values. Can you drop the `=false` from this field?